Description: Prevent SEGV in gs_setdevice_no_erase.
 Prevent SEGV if gsicc_init_iccmanager fails.
 .
 Error created using :-
 MEMENTO_FAILAT=350 ./membin/gpcl6 -sDEVICE=pbmraw -dMaxBitmap=2000
 -o /dev/null ./tests_private/pcl/pcl5cfts/fts.0070
Origin: upstream, http://git.ghostscript.com/?p=ghostpdl.git;h=52a37b6
Author: Shailesh Mistry <shailesh.mistry@hotmail.co.uk>
Forwarded: yes
Bug: http://bugs.ghostscript.com/show_bug.cgi?id=697545
Last-Update: 2018-10-18
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/base/gsdevice.c
+++ b/base/gsdevice.c
@@ -498,7 +498,9 @@
     if (libctx->io_device_table != NULL) {
         cmm_dev_profile_t *dev_profile;
         if (pgs->icc_manager->lab_profile == NULL) {  /* pick one not set externally */
-            gsicc_init_iccmanager(pgs);
+            code = gsicc_init_iccmanager(pgs);
+            if (code < 0)
+                return(code);
         }
         /* Also, if the device profile is not yet set then take care of that
            before we start filling pages, if we can */
@@ -720,7 +722,8 @@
 
         if ((code = gs_setdevice_no_erase(pgs, ndev)) < 0)
             gs_free_object(pgs->memory, ndev, "gs_copydevice(device)");
-        gs_currentdevice_inline(pgs)->LockSafetyParams = saveLockSafety;
+        if (pgs->device != NULL)
+            pgs->device->LockSafetyParams = saveLockSafety;
     }
     return code;
 }
